Delcam.ProductInterface.PowerMILL/Classes/PMSetup.cs

```diff
@@ -47,6 +51,37 @@ internal override string Identifier
             get { return SETUP_IDENTIFIER; }
         }
 
+        /// <summary>
+        /// Gets the list of all the toolpaths that are in this Setup.
+        /// </summary>
+        public List<PMToolpath> Toolpaths
+        {
+            get
+            {
+                // Get the list of toolpaths that are in the Setup
+                var toolpathsGUIDs = PowerMill.DoCommandEx($"PRINT FOLDER 'Setup\\{Name}'").ToString();
+                // Remove by replacement all CRs
+                toolpathsGUIDs = toolpathsGUIDs.Replace(((char)13).ToString(), string.Empty);
+                // Remove by replacement all prefixes
+                toolpathsGUIDs = toolpathsGUIDs.Replace($"Setup\\{Name}\\\\", string.Empty);
+                // Split the collection by NL
+                var splitToolpathGUIDs = toolpathsGUIDs.Split((char)10);
+                // Convert to GUIDs to resolve parsing concerns
+                var targetGUIDS = splitToolpathGUIDs.Select(x => new Guid(x));
+
+                // Use the retrieved GUIDs to select from the session's complete toolpath collection.
+                // Iterating in this manor will order the resultant list by the order in which the 
+                // toolpaths appear in the setup
+                var setupToolpaths = from toolpathGUID in targetGUIDS 
```

Comment:
You've used two different types of linq query between here and line 70. Be consistent and use:
`var setupToolpaths = targetGuids.Select(x => PowerMill.ActiveProject.Toolpaths.First(x.ID == toolpathGUID);`

Thinking about this though, what happens if you can't find a toolpath by the specified id in the list of Toolpaths. e.g. if the cache is out of date. Might be worth wrapping in a try catch and throwing an exception if the toolpath cannot be found?

Reply:
It would need to be:
`var setupToolpaths = targetGUIDs.Select(targetGUID => PowerMill.ActiveProject.Toolpaths.First(toolpath => toolpath.ID == targetGUID));`
Which ends up being a long and ugly line, hense the style change, but I can keep it if you prefer?

I understood that we don't generally handle for the cache being out of date otherwise we'd add checks right down to the indexers on collections. It's easily added if you'd like it? What would I throw? A general `Exception`?

Reply:
Personally I prefer the version. We had a whole vote in my old team and decided we preferred using this kind of notation so I will let that vote carry for now :-)
I think throwing an Exception with a nice explanation would be better than a "Sequence doesn't contain the item" error that a user of the API would be unable to debug. Something like "Failed to populate list of Toolpaths within Setup. The list of toolpaths may be out of date. Consider using Refresh function before reading the list of toolpaths."

Reply:
Roger, Roger. On it :)

Reply:
Requested changes made on #7581227